lib/streamlit/runtime/scriptrunner_utils/script_run_context.py

```diff
@@ -192,15 +201,24 @@ def enqueue(self, msg: ForwardMsg) -> None:
 
         msg.metadata.active_script_hash = self.active_script_hash
 
+        # We populate the hash and cacheable field for all messages.
+        # Besides the forward message cache, the hash might also be used
+        # for other aspects within the frontend.
+        populate_hash_if_needed(msg)
+        msg_to_send = msg
+        if msg.metadata.cacheable and msg.hash and msg.hash in self.cached_messages:
```

Comment:
probably out-of-scope for this PR and could theoretically be a follow-up, but I am wondering whether we could get rid of `cacheable` now since the backend does not store the messages locally anymore. I guess in the past we wanted to prevent to store too many messages on the server-side, but since this is not happening anymore, I am not sure whether there is a good reason to let the frontend decide which messages to cache (as long as the message has a `hash` with it). What do you think?